pkg/iam/mapping.go

```diff
@@ -29,6 +29,33 @@ type Identity interface {
 	Account() string
 }
 
+// CompareIdentity takes 2 Identity values and checks to see if they are identitcal
+func CompareIdentity(a, b Identity) bool {
+	sameAccount := (a.ARN() == b.ARN() &&
+		a.Type() == b.Type() &&
+		a.Username() == b.Username() &&
+		a.Account() == b.Account())
+
+	if !sameAccount {
+		return false
+	}
+
+	aGroups := a.Groups()
+	bGroups := b.Groups()
+	if len(aGroups) != len(bGroups) {
+		return false
+	}
+
+	for i := range aGroups {
+		if aGroups[i] != bGroups[i] {
```

Comment:
Hm, in the description you are saying this:

>This might be more robust if the Identity groups were sorted before being compared. Wasn't sure if it made sense to implement that way.



So if this isn't the case, then you are depending here on a thing that doesn't exist, right? And in fact, checking the code, we do not sort groups. Please add a unit test which checks for any kind of order of checking if these are equal. :)

Reply:
I made it so that it sorts right before the compare.